File: Find First Palindromic String in the Array.py
class Solution:
    def firstPalindrome(self, words: List[str]) -> str:
        for word in words:
            n = len(word)
            i, j = 0, n - 1
            palindrome = True
            while i < j:
                if word[i] == word[j]:
                    i += 1
                    j -= 1
                else:
                    palindrome = False
                    break
            if palindrome:
                return word
        return ""

File: Find First Palindromic String in the Array.cpp
class Solution {
public:
    string firstPalindrome(vector<string>& words) {
        for (string & word: words) {
            int n = word.size();
            int i = 0;
            int j = n-1;
            bool palindrome = true;
            while (i < j) {
                if (word[i] == word[j]) {
                    i++;
                    j--;
                } else {
                    palindrome = false;
                    break;
                }
            }
            if (palindrome) return word;
        }
        return "";
    }
};
